tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -41,13 +41,42 @@ type Report struct {
 	Flaky []string `json:"flaky"`
 }
 
+// JSONClient contains the set of operations a JSON reporter needs
+type Client interface {
+	CreateReportForRepo(repo string, flaky []string, writeFile bool) (*Report, error)
+	GetFlakyTests(repo string) ([]string, error)
+	GetReportRepos() ([]string, error)
+	GetFlakyTestReport(repo string, buildID int) ([]*Report, error)
+}
+
+// Client is simply a way to call methods, it does not contain any data itself
+type JSONClient struct{}
```

Comment:
add verification for implementation
`var _ Client = (*JSONClient)(nil)`

Reply:
done